frontend/src/components/tables/table-view-column-menu.tsx

```diff
@@ -384,3 +402,128 @@ function TableColumnEditDialog({
     </Dialog>
   )
 }
+
+function TableColumnIndexDialog({
+  tableId,
+  column,
+  open,
+  onOpenChange,
+}: {
+  tableId?: string
+  column: TableColumnRead
+  open: boolean
+  onOpenChange: () => void
+}) {
+  const { workspaceId } = useWorkspace()
+  const { updateColumn, updateColumnIsPending } = useUpdateColumn()
+
+  if (!tableId || !workspaceId || !column.is_index) {
```

Comment:
```suggestion
  if (!tableId || !workspaceId) {
```
I realize I misled you a bit here, looks like this column won't show anything if the index hasn't been created